### ISSUE TITLE
🐛 [firebase_messaging] Cocoapods unable to find compaitable version of Firebase/Messaging

### ISSUE THREAD

**Issue by NebulaFox** [FirebaseExtended/flutterfire#8135]
## Bug report

**Describe the bug**
Unable to generate an ios app that uses firebase messaging, due to cocoa pod dependency errors.

```
[!] CocoaPods could not find compatible versions for pod "Firebase/Messaging":
      In snapshot (Podfile.lock):
        Firebase/Messaging (= 8.9.0)

      In Podfile:
        firebase_messaging (from `.symlinks/plugins/firebase_messaging/ios`) was resolved to 11.2.4, which depends on
          Firebase/Messaging (= 8.11.0)

    Specs satisfying the `Firebase/Messaging (= 8.9.0), Firebase/Messaging (= 8.11.0)` dependency were found, but they required a higher minimum deployment
    target.
```

### Steps to reproduce

1. Add dependecy `firebase_messaging: 11.2.6`

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel beta, 2.11.0-0.1.pre, on macOS 12.2.1 21D62 darwin-arm, locale en-GB)
    • Flutter version 2.11.0-0.1.pre at /usr/local/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b101bfe32f (5 days ago), 2022-02-16 07:36:54 -0800
    • Engine revision e355993572
    • Dart version 2.17.0 (build 2.17.0-69.2.beta)
    • DevTools version 2.10.0-dev.1
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Changing current working directory to: /Users/robbiebykowski/Dropbox/Apps/2021/PetId
Dart SDK 2.17.0-69.2.beta
Flutter SDK 2.11.0-0.1.pre
PetId 2.1.0-pre+13

dependencies:
- adjust_sdk 4.29.1 [flutter meta]
- async 2.8.2 [collection meta]
- firebase_analytics 9.1.0 [firebase_analytics_platform_interface firebase_analytics_web firebase_core firebase_core_platform_interface flutter]
- firebase_core 1.12.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_crashlytics 2.5.0 [firebase_core firebase_core_platform_interface firebase_crashlytics_platform_interface flutter stack_trace]
- firebase_messaging 11.2.4 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- flutter 0.0.0 [characters collection material_color_utilities meta typed_data vector_math sky_engine]
- flutter_secure_storage 5.0.2 [meta flutter flutter_secure_storage_linux flutter_secure_storage_macos flutter_secure_storage_platform_interface flutter_secure_storage_web flutter_secure_storage_windows]
- flutter_svg 1.0.3 [flutter meta path_drawing vector_math xml]
- http 0.13.4 [async http_parser meta path]
- image_picker 0.8.4+6 [flutter flutter_plugin_android_lifecycle image_picker_for_web image_picker_platform_interface]
- intl_phone_number_input 0.7.0+2 [flutter meta libphonenumber_plugin equatable collection]
- logging 1.0.2
- lottie 1.2.1 [archive characters flutter path vector_math]
- mailto 2.0.0
- material_design_icons_flutter 5.0.6595 [flutter]
- pin_code_fields 7.3.0 [flutter]
- provider 6.0.2 [collection flutter nested]
- share 2.0.4 [meta mime flutter]
- sliver_tools 0.2.5 [flutter]
- sprintf 6.0.0
- url_launcher 6.0.18 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]

dev dependencies:
- build_runner 2.1.7 [args async analyzer build build_config build_daemon build_resolvers build_runner_core code_builder collection crypto dart_style frontend_server_client glob graphs http_multi_server io js logging meta mime package_config path pool pub_semver pubspec_parse shelf shelf_web_socket stack_trace stream_transform timing watcher web_socket_channel yaml]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph typed_data]
- integration_test 0.0.0 [flutter flutter_driver flutter_test path vm_service archive async boolean_selector characters charcode clock collection crypto fake_async file matcher material_color_utilities meta source_span stack_trace stream_channel string_scanner sync_http term_glyph test_api typed_data vector_math webdriver]
- mockito 5.0.17 [analyzer build code_builder collection dart_style matcher meta path source_gen test_api]
- test 1.20.1 [analyzer async boolean_selector collection coverage http_multi_server io js node_preamble package_config path pool shelf shelf_packages_handler shelf_static shelf_web_socket source_span stack_trace stream_channel typed_data web_socket_channel webkit_inspection_protocol yaml test_api test_core]

dependency overrides:
- libphonenumber 2.0.2 [flutter meta]

transitive dependencies:
- _fe_analyzer_shared 31.0.0 [meta]
- analyzer 2.8.0 [_fe_analyzer_shared cli_util collection convert crypto glob meta package_config path pub_semver source_span watcher yaml]
- archive 3.1.8 [crypto path]
- args 2.3.0
- boolean_selector 2.1.0 [source_span string_scanner]
- build 2.2.1 [analyzer async convert crypto glob logging meta path]
- build_config 1.0.0 [checked_yaml json_annotation path pubspec_parse yaml]
- build_daemon 3.0.1 [built_collection built_value http_multi_server logging path pool shelf shelf_web_socket stream_transform watcher web_socket_channel]
- build_resolvers 2.0.6 [analyzer async build crypto graphs logging path package_config pool pub_semver stream_transform yaml]
- build_runner_core 7.2.3 [async build build_config build_resolvers collection convert crypto glob graphs json_annotation logging meta path package_config pool timing watcher yaml]
- built_collection 5.1.1
- built_value 8.1.4 [built_collection collection fixnum meta]
- characters 1.2.0
- charcode 1.3.1
- checked_yaml 2.0.1 [json_annotation source_span yaml]
- cli_util 0.3.5 [meta path]
- clock 1.1.0
- code_builder 4.1.0 [built_collection built_value collection matcher meta]
- collection 1.15.0
- convert 3.0.1 [typed_data]
- coverage 1.0.3 [args logging package_config path source_maps stack_trace vm_service]
- cross_file 0.3.2 [flutter js meta]
- crypto 3.0.1 [collection typed_data]
- dart_style 2.2.1 [analyzer args path pub_semver source_span]
- equatable 2.0.3 [collection meta]
- fake_async 1.2.0 [clock collection]
- file 6.1.2 [meta path]
- firebase_analytics_platform_interface 3.0.5 [firebase_core flutter meta plugin_platform_interface]
- firebase_analytics_web 0.4.0+6 [firebase_analytics_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- firebase_core_platform_interface 4.2.4 [collection flutter meta plugin_platform_interface]
- firebase_core_web 1.5.4 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_crashlytics_platform_interface 3.1.13 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_platform_interface 3.1.6 [firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 2.2.7 [firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins js meta]
- fixnum 1.0.0
- flutter_driver 0.0.0 [file flutter flutter_test fuchsia_remote_debug_protocol path meta vm_service webdriver archive async boolean_selector characters charcode clock collection crypto matcher material_color_utilities platform process source_span stack_trace stream_channel string_scanner sync_http term_glyph test_api typed_data vector_math]
- flutter_plugin_android_lifecycle 2.0.5 [flutter]
- flutter_secure_storage_linux 1.1.0 [flutter flutter_secure_storage_platform_interface]
- flutter_secure_storage_macos 1.1.0 [flutter flutter_secure_storage_platform_interface]
- flutter_secure_storage_platform_interface 1.0.0 [flutter plugin_platform_interface]
- flutter_secure_storage_web 1.0.2 [flutter flutter_web_plugins flutter_secure_storage_platform_interface js]
- flutter_secure_storage_windows 1.1.2 [flutter flutter_secure_storage_platform_interface]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta typed_data vector_math]
- frontend_server_client 2.1.2 [async path]
- fuchsia_remote_debug_protocol 0.0.0 [process vm_service file meta path platform]
- glob 2.0.2 [async collection file path string_scanner]
- graphs 2.1.0 [collection]
- http_multi_server 3.0.1 [async]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- image_picker_for_web 2.1.5 [flutter flutter_web_plugins image_picker_platform_interface pedantic]
- image_picker_platform_interface 2.4.3 [flutter http plugin_platform_interface cross_file]
- io 1.0.3 [meta path string_scanner]
- js 0.6.4
- json_annotation 4.4.0 [meta]
- libphonenumber_platform_interface 0.3.1 [flutter plugin_platform_interface]
- libphonenumber_plugin 0.2.3 [flutter flutter_web_plugins libphonenumber_platform_interface libphonenumber_web libphonenumber]
- libphonenumber_web 0.2.0+1 [flutter flutter_web_plugins js libphonenumber_platform_interface]
- matcher 0.12.11 [stack_trace]
- material_color_utilities 0.1.3
- meta 1.7.0
- mime 1.0.1
- nested 1.0.0 [flutter]
- node_preamble 2.0.1
- package_config 2.0.2 [path]
- path 1.8.1
- path_drawing 1.0.0 [vector_math meta path_parsing flutter]
- path_parsing 1.0.0 [vector_math meta]
- pedantic 1.11.1
- petitparser 4.4.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.2 [meta]
- pool 1.5.0 [async stack_trace]
- process 4.2.4 [file path platform]
- pub_semver 2.1.0 [collection meta]
- pubspec_parse 1.2.0 [checked_yaml collection json_annotation pub_semver yaml]
- shelf 1.2.0 [async collection http_parser path stack_trace stream_channel]
- shelf_packages_handler 3.0.0 [path shelf shelf_static]
- shelf_static 1.1.0 [convert http_parser mime path shelf]
- shelf_web_socket 1.0.1 [shelf stream_channel web_socket_channel]
- sky_engine 0.0.99
- source_gen 1.2.1 [analyzer async build dart_style glob meta path source_span yaml]
- source_map_stack_trace 2.1.0 [path stack_trace source_maps]
- source_maps 0.10.10 [source_span]
- source_span 1.8.1 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- stream_transform 2.0.0
- string_scanner 1.1.0 [charcode source_span]
- sync_http 0.3.0
- term_glyph 1.2.0
- test_api 0.4.9 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- test_core 0.4.11 [analyzer async args boolean_selector collection coverage frontend_server_client glob io meta package_config path pool source_map_stack_trace source_maps source_span stack_trace stream_channel vm_service yaml matcher test_api]
- timing 1.0.0 [json_annotation]
- typed_data 1.3.0 [collection]
- url_launcher_android 6.0.14 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.0.14 [flutter url_launcher_platform_interface]
- url_launcher_linux 2.0.3 [flutter]
- url_launcher_macos 2.0.3 [flutter]
- url_launcher_platform_interface 2.0.5 [flutter plugin_platform_interface]
- url_launcher_web 2.0.6 [flutter flutter_web_plugins url_launcher_platform_interface]
- url_launcher_windows 2.0.2 [flutter]
- vector_math 2.1.1
- vm_service 7.5.0
- watcher 1.0.1 [async path]
- web_socket_channel 2.1.0 [async crypto stream_channel]
- webdriver 3.0.0 [archive matcher path stack_trace sync_http]
- webkit_inspection_protocol 1.0.0 [logging]
- xml 5.3.1 [collection meta petitparser]
- yaml 3.1.0 [collection source_span string_scanner]
```

</details>

---

**Comment by darshankawar** [comment]
@NebulaFox 

As the error indicates below:

```
In snapshot (Podfile.lock):
        Firebase/Messaging (= 8.9.0)

      In Podfile:
        firebase_messaging (from `.symlinks/plugins/firebase_messaging/ios`) was resolved to 11.2.4, which depends on
          In snapshot (Podfile.lock):
        Firebase/Messaging (= 8.9.0)

      In Podfile:
        firebase_messaging (from `.symlinks/plugins/firebase_messaging/ios`) was resolved to 11.2.4, which depends on
          Firebase/Messaging (= 8.11.0)
```

Your podfile.lock has plugin version dependency on 8.9.0, but you seem to have upgraded the `firebase_messaging` plugin to 11.2.4 which probably requires `8.11.0` in `podfile` and `podfile.lock`, so either you can run `pod repo update` or delete `podfile.lock` and regenerate it which should pick up `Firebase/Messaging (= 8.11.0)`. You can also run `flutter pub cache repair` .

I'll go ahead and close this as above error occured due to podfile.lock is pointing to older version of the pod. Clearing or recreating it should resolve the issue. If not, write in comments and I'll reopen it.

**Comment by darshankawar** [comment]
@NebulaFox 

As the error indicates below:

```
In snapshot (Podfile.lock):
        Firebase/Messaging (= 8.9.0)

      In Podfile:
        firebase_messaging (from `.symlinks/plugins/firebase_messaging/ios`) was resolved to 11.2.4, which depends on
          In snapshot (Podfile.lock):
        Firebase/Messaging (= 8.9.0)

      In Podfile:
        firebase_messaging (from `.symlinks/plugins/firebase_messaging/ios`) was resolved to 11.2.4, which depends on
          Firebase/Messaging (= 8.11.0)
```

Your podfile.lock has plugin version dependency on 8.9.0, but you seem to have upgraded the `firebase_messaging` plugin to 11.2.4 which probably requires `8.11.0` in `podfile` and `podfile.lock`, so either you can run `pod repo update` or delete `podfile.lock` and regenerate it which should pick up `Firebase/Messaging (= 8.11.0)`. You can also run `flutter pub cache repair` .

I'll go ahead and close this as above error occured due to podfile.lock is pointing to older version of the pod. Clearing or recreating it should resolve the issue. If not, write in comments and I'll reopen it.
